src/nifs/protogalaxy/mod.rs

```diff
@@ -425,6 +429,96 @@ impl<C: CurveAffine, const L: usize> FoldingScheme<C, L> for ProtoGalaxy<C, L> {
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum VerifyError<F: PrimeField> {
+    #[error("Error while evaluate witness: {0:?}")]
+    PlonkEval(plonk::eval::Error),
+    #[error("Expected `e` {expected_e:?}, but evaluated is {evaluated_e:?}")]
+    MismatchE { expected_e: F, evaluated_e: F },
+    #[error("Permutation check failed")]
+    Perm(plonk::Error),
+}
+
+impl<C: CurveAffine, const L: usize> IsSatAccumulator<C, L> for ProtoGalaxy<C, L> {
+    type VerifyError = VerifyError<C::ScalarExt>;
+
+    fn is_sat_acc(
+        _ck: &CommitmentKey<C>,
```

Comment:
I think there is commitment-witness check need to be added, as well as the perm check.

Reply:
#340

Reply:
approved #340 
Still need add back is_sat_perm for is_sat_acc though...

Reply:
I added the `is_sat` function with all the checks. `is_sat_acc` - checks exactly the accumulation - checks exactly the accumulation, not the accumulator

It might be worth adding a rustdoc about it. will fix

Reply:
I've split into two traits (now the second one can't be mistakenly implemented by itself, it will only be implemented through the first trait) and added comments. Please look at the second PR again
е